src/rpc/net.cpp

```diff
@@ -768,7 +754,21 @@ static RPCHelpMan listbanned()
 
     banmap_t banMap;
     node.banman->GetBanned(banMap);
-
+    if (!request.params[0].isNull()) {
+        CNetAddr netAddr;
+        LookupHost(request.params[0].get_str(), netAddr, false);
+        if (!netAddr.IsValid()) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Error: Invalid ip address");
+        }
+        for (auto it = banMap.begin(); it != banMap.end();) {
+            CSubNet ban_sub_net = it->first;
+            if (!ban_sub_net.Match(netAddr)) {
+                it = banMap.erase(it);
+            } else {
+                it++;
```

Comment:
38abc322 prefer prefix iterator
```suggestion
                ++it;
```

Reply:
Done